### PR TITLE
Use GeomAPI_Interpolate for BSpline construction

### PR DESCRIPTION
**Pull request by adam-urbanczyk**
This allows to specify tangents and create closed bspline curves

Solves #90 

* Spline with/without constraints 
afbeelding

* Closed spline (converted to Face) 
afbeelding

### CODE REVIEW

**Comment by adam-urbanczyk** [comment]
NB: I also refactored `tangentAt` for arbitrary location (in [0,1] parameter space) on an edge. This is used in the tests for the spline functionality.

**Comment by jmwright** [comment]
@adam-urbanczyk I won't get a chance to look at this until the morning (US time), but I'll review it first thing.

File: cadquery/occ_impl/shapes.py [hunk 1]
```diff
@@ -729,18 +730,28 @@ def makeCircle(cls, radius, pnt=Vector(0, 0, 0), dir=Vector(0, 0, 1), angle1=360
...
         Interpolate a spline through the provided points.
         :param cls:
         :param listOfVector: a list of Vectors that represent the points
+        :param tangents: tuple of Vectors specifying start and finish tangent
+        :param periodic: creation of peridic curves
+        :param tol: tolerance of the algorithm (consult OCC documentation)
         :return: an Edge
         """
-        pnts = TColgp_Array1OfPnt(0, len(listOfVector) - 1)
+        pnts = TColgp_HArray1OfPnt(1, len(listOfVector))
```
Comment (jmwright) [inline cadquery/occ_impl/shapes.py]: Odd that this array isn't 0 based, but it looks like they did it intentionally.

File: tests/TestCadQuery.py [hunk 2]
```diff
@@ -362,6 +362,32 @@ def testRevolveCone(self):
...
+        # Spline path - just a smoke test
+        path = Workplane("XZ").spline(pts).val()
+
+        # Closed spline
+        path_closed = Workplane("XZ").spline(pts,periodic=True).val()
+        self.assertTrue(path_closed.IsClosed())
+        
+        # attempt to build a face
+        w = Wire.assembleEdges([path_closed,])
+        f = Face.makeFromWires(w)
```
Comment (jmwright) [inline tests/TestCadQuery.py]: Do you want to add an assert here to make sure that the face was built correctly?

Comment (adam-urbanczyk) [reply tests/TestCadQuery.py]: Yes, good catch! Actually the `isValid` method was not implemented. Now I check if the generated face is valid.

File: cadquery/occ_impl/shapes.py (hunk 1 above)
Comment (adam-urbanczyk) [reply cadquery/occ_impl/shapes.py]: Yes, it is weird. I could only get `GeomAPI_Interpolate` to work with 1-based arrays.

**Pull request merged**